internal/services/automation/automation_watcher_test.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type WatcherResource struct{}
+
+func (a WatcherResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.WatcherID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Automation.WatcherClient.Get(ctx, id.ResourceGroup, id.AutomationAccountName, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving Type %s: %+v", id, err)
+	}
+	return utils.Bool(resp.WatcherProperties != nil), nil
+}
+
+func (a WatcherResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-auto-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_automation_account" "test" {
+  name                = "acctestAA-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku_name            = "Basic"
+}
+
+resource "azurerm_automation_credential" "test" {
+  name                    = "acctest-%[1]d"
+  resource_group_name     = azurerm_resource_group.test.name
+  automation_account_name = azurerm_automation_account.test.name
+  username                = "test_user"
+  password                = "test_pwd"
+}
+
+resource "azurerm_automation_hybrid_runbook_worker_group" "test" {
+  resource_group_name     = azurerm_resource_group.test.name
+  automation_account_name = azurerm_automation_account.test.name
+  name                    = "acctest-%[1]d"
+  credential_name         = azurerm_automation_credential.test.name
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestnw-%[1]d"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_network_interface" "test" {
+  name                = "acctni-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  ip_configuration {
+    name                          = "testconfiguration1"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "test" {
+  name                = "acctestVM-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_G5"
```

Comment:
Do we need to use this size?

```
------- Stdout: -------
=== RUN   TestAccWatcher_basic
=== PAUSE TestAccWatcher_basic
=== CONT  TestAccWatcher_basic
    testcase.go:110: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating Linux Virtual Machine: (Name "acctestVM-220906160028953594" / Resource Group "acctestRG-auto-220906160028953594"): compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="SkuNotAvailable" Message="The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_G5' is currently not available in location 'westus2'. Please try another size or deploy to a different location or different zone. See https://aka.ms/azureskunotavailable for details."
        
          with azurerm_linux_virtual_machine.test,
          on terraform_plugin_test.tf line 62, in resource "azurerm_linux_virtual_machine" "test":
          62: resource "azurerm_linux_virtual_machine" "test" {
        
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Network Interface: (Name "acctni-220906160028953594" / Resource Group "acctestRG-auto-220906160028953594"): network.InterfacesClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="NicReservedForAnotherVm" Message="Nic(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another nic(s) or retry after 180 seconds. Reserved VM: /subscriptions/*******/resourceGroups/acctestRG-auto-220906160028953594/providers/Microsoft.Compute/virtualMachines/acctestVM-220906160028953594" Details=[]
        
--- FAIL: TestAccWatcher_basic (55.70s)
FAIL
```

Reply:
yes, it's a required property. the value of size should be related to the subscription/account maybe. in my local env, this test passed. and I changed it to the same as other VM acctests value `Standard_F2`

Reply:
![image](https://user-images.githubusercontent.com/2633022/189022716-549bce59-15d7-434e-8adf-d2467574a8fd.png)

now all tests passed in tc